src/transform/plugins/links/index.ts

```diff
@@ -241,14 +236,19 @@ const index: MarkdownItPluginCb<ProcOpts & Options> = (md: MarkdownItIncluded, o
                 let j = 0;
 
                 while (j < childrenTokens.length) {
-                    const isLinkOpenToken = childrenTokens[j].type === 'link_open';
-                    const tokenClass = childrenTokens[j].attrGet('class');
+                    const token = childrenTokens[j];
+
+                    const isLinkOpenToken = token.type === 'link_open';
+                    const tokenClass = token.attrGet('class');
 
                     /*  Don't process anchor links */
                     const isYfmAnchor = tokenClass ? tokenClass.includes('yfm-anchor') : false;
+                    const wasProcessedBefore = Boolean(token.meta?.yfmLinkPluginProcessed);
 
-                    if (isLinkOpenToken && !isYfmAnchor) {
+                    if (isLinkOpenToken && !wasProcessedBefore && !isYfmAnchor) {
                         processLink(md, state, childrenTokens, j, opts);
+
+                        (token.meta ??= {}).yfmLinkPluginProcessed = true;
```

Comment:
Looks like some ancient curse from lisp or something. Can this be expressed cleaner?  Like an `if` would be much easier to read.

Reply:
Ha, I knew this would raise some eyebrows. I will fix it then